tools/update_certs.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python3
+# Copyright 2024 The Cobalt Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tool for updating Cobalt SSL Certificates.
+
+Usage:
+  update_certs.py -r <cobalt_repo> -b <openssl_bin> -p <pem_files>
+
+Args:
+    cobalt_repo: The path to the Cobalt repository root.
+    openssl_bin: The path to the OpenSSL binary, defaults to "/usr/bin/openssl"
+      when not specified.
+    pem_files: The paths to the source pem files.
+"""
+
+import argparse
+import os
+import shutil
+import subprocess
+
+from typing import Set
+
+_CERTS_PATH = 'cobalt/content/ssl/certs'
+_BUILD_PATH = 'cobalt/network/certs.gni'
+
+
+def SplitCerts(concatenated_cert_file: str):
+  """Split concatenated .pem file into a list of individual certs."""
+  begin_anchor = '-----BEGIN CERTIFICATE-----'
+  end_anchor = '-----END CERTIFICATE-----'
+
+  cert_list = []
+  current_cert = ''
+  in_cert = False
+
+  for line in concatenated_cert_file:
+    if not in_cert and line.find(begin_anchor) != -1:
+      in_cert = True
+      current_cert += line
+    elif in_cert and line.find(end_anchor) != -1:
+      in_cert = False
+      current_cert += line
+      cert_list.append(current_cert)
+      current_cert = ''
+    elif in_cert:
+      current_cert += line
+
+  return cert_list
+
+
+def ComputeCertFilename(openssl_bin: str, cert: str):
+  """Use x509 to get the hashed filename for the individual cert."""
+  command_args = [openssl_bin, 'x509', '-subject_hash', '-noout']
+  output = subprocess.check_output(command_args, input=cert.encode('utf-8'))
+
+  return output.decode().rstrip() + '.0'
+
+
+def UpdateBuild(build_file: str, cert_filepaths: Set[str]):
+  cert_filepaths = list(cert_filepaths)
+  cert_filepaths.sort()
+
+  # Removes existing build file
+  if os.path.exists(build_file):
```

Comment:
I don't think this is necessary, doesn't opening the file in write mode just overwrite it?

Reply:
done.